decode/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use base64::Engine;
+use clap::{Parser, Subcommand};
+use proteus_wasm::internal::message::SessionTag;
+use proteus_wasm::internal::util::fmt_hex;
+use proteus_wasm::keys::{PreKeyBundle, Signature};
+use proteus_wasm::message::{CipherMessage, Envelope, Message, PreKeyMessage};
+use std::ops::Deref;
+
+#[derive(Debug)]
+#[allow(dead_code)]
+struct ProteusPreKeyBundle {
+    pub version: u8,
+    pub prekey_id: u16,
+    pub public_key: String,
+    pub identity_key: String,
+    pub signature: Option<Signature>,
+}
+
+impl From<PreKeyBundle> for ProteusPreKeyBundle {
+    fn from(bundle: PreKeyBundle) -> Self {
+        Self {
+            version: bundle.version,
+            prekey_id: bundle.prekey_id.value(),
+            public_key: bundle.public_key.fingerprint(),
+            identity_key: bundle.identity_key.fingerprint(),
+            signature: bundle.signature,
+        }
+    }
+}
+
+#[derive(Debug)]
+#[allow(dead_code)]
+struct ProteusEnvelope {
+    version: u16,
+    mac: String,
+    message: ProteusMessage,
+}
+
+#[derive(Debug)]
+#[allow(dead_code)]
+enum ProteusMessage {
+    Plain(ProteusCipherMessage),
+    Keyed(ProteusPrekeyMessage),
```

Comment:
At first glance, the variant names look a bit confusing (without knowing any details about the underlying types). Is there a particular reason for `Plain` and `Keyed`? Variant names like `Cipher` and `Prekey` would more closely hint the wrapped type